### PR TITLE
Allow injection of types not just mapping

### PR DESCRIPTION
**Pull request by jadb**
This would allow to have configurable types and be able to move some logic currently
done at the `Behavior` level instead but would fit the Type much more (i.e. data encryption)

### CODE REVIEW

**Comment by markstory** [comment]
Could we get some tests to make sure this change doesn't break in the future.

**Comment by jadb** [comment]
Sure!

**Comment by jadb** [comment]
@markstory, I think this is good to go now.

File: tests/TestCase/Database/TypeTest.php [hunk 1]
```diff
@@ -138,6 +138,12 @@ public function testMapAndBuild()
         $this->assertInstanceOf($fooType, $type);
         $this->assertEquals('foo', $type->getName());
         $this->assertEquals('text', $type->getBaseType());
+
+        $fooType = new FooType();
+        Type::map('foo2', $fooType);
+        $map = Type::map();
+        $this->assertEquals($fooType, $map['foo2']);
+        $this->assertEquals($fooType, Type::map('foo2'));
```
Comment (lorenzo) [inline tests/TestCase/Database/TypeTest.php]: Should be assertSame()

Comment (jadb) [inline tests/TestCase/Database/TypeTest.php]: Fixing now

**Pull request merged**